EazyE2E/Process/EzProcess.cs

```diff
@@ -1,42 +1,69 @@
+        public int ProcessId => _process.Id;
+        public string Arguments { get; set; } = string.Empty;
+        public ProcessWindowStyle WindowStyle { get; set; } = ProcessWindowStyle.Normal;
+
+        public EzProcess(string processFullPath, string processName)
         {
+            _processFullPath = processFullPath;
             _processName = processName;
-            _arguments = arguments;
-            _winStyle = winStyle;
         }
 
-        public int StartProcess()
+
+        public void StartProcess()
         {
             var start = new ProcessStartInfo
             {
-                Arguments = _arguments,
-                FileName = _processName,
-                WindowStyle = _winStyle
+                FileName = _processFullPath,
+                WindowStyle = WindowStyle,
+                Arguments = Arguments
             };
-            int exitCode;
 
-            using (System.Diagnostics.Process proc = System.Diagnostics.Process.Start(start))
-            {
-                if (proc == null) throw new NullReferenceException($"Could not find process where ProcessName = {_processName}, and Arguments = {_arguments}");
-                proc.Start(); //todo not sure if this is the right method call.  Want to go to the next line once the process is fully launched and ready to begin testing
-                exitCode = proc.ExitCode;
-            }
+            _process = System.Diagnostics.Process.Start(start);
+
+            FindRunningProcess();
+        }
 
-            return exitCode;
+        /// <summary>
+        /// If a process is already started when calling Process.Start,
+        /// and the process looks for an existing process, sometimes
+        /// it will forward the request to the existing process and then
+        /// exit.  This method handles that scenario and gets a reference
+        /// to the correct process.
+        /// </summary>
+        private void FindRunningProcess()
+        {
+            // Give the process 1 second to start or forward the start request.
+            _process.WaitForExit(1000);
+
+            // If it hasn't exited, we've probably
+            // got the right process.
+            if (!_process.HasExited) return;
+
+            // If it has exited, find the currently running process.
+            var process = System.Diagnostics.Process.GetProcessesByName(_processName);
+            if (process.Length > 1) throw new InvalidOperationException($"There are more than one process with the name {_processName}.  Please close all running processes with that name.");
+            if (process.Length == 0) throw new InvalidOperationException($"A process with the name {_processName} could not be found.");
+            _process = process.Single();
         }
 
-        public string ProcessName => _processName;
-        public string Arguments => _arguments;
+        public void Dispose()
```

Comment:
Close the process upon disposing of the EzProcess object.  This is the implementation if the IDisposable interface.